Gemfile

```
@@ -23,6 +23,12 @@ gem 'puppet', *location_for(ENV['PUPPET_VERSION']) if ENV['PUPPET_VERSION']
 
 group :test do
   gem 'rspec'
+  # dependency for Puppet on Ruby 3.4
+  gem 'base64', '~> 0.2'
+  gem 'getoptlong', '~> 0.2'
+  gem 'syslog', '~> 0.3'
+  # dependency for Puppet on Ruby 3.3
+  gem 'racc', '~> 1.8', '>= 1.8.1'
```

Perhaps codify that and make it simpler?
```suggestion
  gem 'racc', '~> 1.8' if RUBY_VERSION >= '3.3'
```

And I'd be tempted to do the same for 3.4 above.